cpp/ql/src/experimental/Security/CWE/CWE-266/IncorrectPrivilegeAssignment.ql

```diff
@@ -72,12 +73,14 @@ where
       globalValueNumber(fc.getArgument(0)) = globalValueNumber(fctmp.getArgument(1)) and
       fc.getArgument(0).getValue() != "0"
     ) and
-    msg = "not use equal argument in umask and " + fctmp.getTarget().getName() + " functions"
+    msg = "Not use equal argument in umask and $@ functions." and
+    fcsnd = fctmp
   )
   or
   exists(ContainsArithmetic exptmp, int i |
     numberArgumentModFunctions(fc.getTarget(), i) and
     globalValueNumber(exptmp) = globalValueNumber(fc.getArgument(i)) and
-    msg = "Using arithmetic to compute the mask may not be safe."
+    msg = "Using arithmetic to compute the mask in $@ may not be safe." and
+    fcsnd = fc
   )
-select fc, msg
+select fc, msg, fcsnd, fcsnd.getTarget().getName()
```

Comment:
This is no longer a valid select clause for a query of kind `problem`. It should select two columns (one code element and one string). See https://codeql.github.com/docs/writing-codeql-queries/about-codeql-queries/#select-clause for more information

Reply:
maybe I misunderstood something.
but links and placeholders are allowed in your description.
```
You can modify the alert message defined in the final column of the select statement to give more detail about the alert or path found by the query using links and placeholders. For more information, see “Defining the results of a query.”
```
could you please explain in more detail what is causing the problem and what needs to be done to fix it

Reply:
This is using the `$@` pattern, see https://codeql.github.com/docs/writing-codeql-queries/defining-the-results-of-a-query/#adding-a-link-to-the-similar-file.

Reply:
Oh! You're absolutely right. I missed the fact that `msg` contained placeholder markers. Never mind!